lib/sr.py

```diff
@@ -130,10 +131,19 @@ def main_host(self):
                 self._main_host = self.pool.get_host_by_uuid(self.hosts_uuids()[0])
         return self._main_host
 
+    def type(self):
+        return self.pool.master.xe('sr-param-get', {'uuid': self.uuid, 'param-name': 'type'})
+
     def content_type(self):
         return self.pool.master.xe('sr-param-get', {'uuid': self.uuid, 'param-name': 'content-type'})
 
     def is_shared(self):
         if self._is_shared is None:
             self._is_shared = self.pool.master.xe('sr-param-get', {'uuid': self.uuid, 'param-name': 'shared'})
         return self._is_shared
+
+    def create_vdi(self, name_label, virtual_size=64):
+        vdi_uuid = self.pool.master.xe('vdi-create', {
+            'name-label': name_label, 'virtual-size': str(virtual_size), 'sr-uuid': self.uuid
```

Comment:
It could be good to add the name prefix we already use for various other objects to the name-label.